package/support/package_darwin.sh

```diff
@@ -197,18 +173,31 @@ if [ "${SIGN_CODE}" = "1" ]; then
 </dict>
 </plist>
 EOF
+
     echo "Validating plist format..."
     plutil -lint entitlements.plist ||
         fail "Entitlements plist file format is invalid"
+
     echo "Signing all substrate executables..."
-    find "${SUBSTRATE_DIR}" -type f -perm +0111 -exec codesign --timestamp --options=runtime --entitlements entitlements.plist -s "${CODE_SIGN_IDENTITY}" {} \; ||
-        fail "Failure while signing executables"
+    for f in $( find "${SUBSTRATE_DIR}"  -type f -exec bash -c "file {} | grep 'Mach-O 64-bit executable'" \; | cut -d ":" -f 1 | cut -d " " -f 1 | uniq )
```

Comment:
The `find` command will need to be quoted to prevent any spaces from breaking up a single path. Might also be a little more clear to pull the file check out into the loop:

```sh
for f in "$( find "${SUBSTRATE_DIR}" -type f -perm +0111 )"; do
  if [[ "$(file "${f}")" != *"Mach-O"* ]]; then
    continue
  fi
  ...
```

Reply:
quoting the `find` command doesn't allow each file to be operated on individually. So it raises an error from signore like

```msg="Error: file could not be read, open /opt/vagrant/bin/vagrant\n/opt/vagrant/embedded/bin/c_rehash\n/opt/vagrant/embedded/bin/xmllint\n/opt/vagrant/embedded/bin/bsdcpio\n/opt/vagrant/embedded/bin/xsltproc\n/opt/vagrant/embedded/bin/iconv\n/opt/vagrant/embedded/bin/xslt-config\n/opt/vagrant/embedded/bin/irb\n/opt/vagrant/embedded/bin/bsdtar\n/opt/vagrant/embedded/bin/rake\n/opt/vagrant/embedded/bin/bundle\n/opt/vagrant/embedded/bin/xml2-config\n/opt/vagrant/embedded/bin/ri\n/opt/vagrant/embedded/bin/curl-config\n/opt/vagrant/embedded/bin/curl\n/opt/vagrant/embedded/bin/racc2y\n/opt/vagrant/embedded/bin/racc\n/opt/vagrant/embedded/bin/bsdcat\n/opt/vagrant/embedded/bin/y2racc\n/opt/vagrant/embedded/bin/erb\n/opt/vagrant/embedded/bin/openssl\n/ ...```